gradio/blocks.py

```diff
@@ -616,19 +616,37 @@ def preprocess_data(self, fn_index, raw_input, state):
             processed_input = raw_input
         return processed_input
 
-    async def call_function(self, fn_index, processed_input):
+    async def call_function(self, fn_index, processed_input, generator=None):
         """Calls and times function with given index and preprocessed input."""
         block_fn = self.fns[fn_index]
-
+        is_generating = False
         start = time.time()
+
         if inspect.iscoroutinefunction(block_fn.fn):
             prediction = await block_fn.fn(*processed_input)
         else:
             prediction = await anyio.to_thread.run_sync(
                 block_fn.fn, *processed_input, limiter=self.limiter
             )
+
+        if inspect.isgeneratorfunction(block_fn.fn):
+            try:
+                if generator is None:
+                    generator = prediction
+                prediction = next(generator)
```

Comment:
Just realized we can't support stateful demos because the updated state is not passed into the next iteration. Would be fine for stable diffusion but that may be relevant for RL.

Happy to leave it out of scope for now since I think all we would need to do to support it is add some logic to the backend to turn the function into a general iterator instead of a generator but might be good to think that through prior to committing to this api in case it may not be general enough. 

Sample stateful demo:

```python
import gradio as gr
import time

def cumulative_sum(up_to, current=0):
    for i in range(up_to):
        time.sleep(1)
        new = i + current
        yield new, new

gr.Interface(cumulative_sum, inputs=[gr.Number(precision=0), "state"],
             outputs=[gr.Number(), "state"]).launch()
```

Reply:
Good catch, will think about this some more!